### PR TITLE
fix: Prevent OSCR score from popping out of dev card with long usernames

### LINKED ISSUE #3981: Bug: Dev Card with long username cause OSCR to pop out of card
A dev card with long username causes the OSCR score to pop out of card. You can see an example of this on Josh's dev card page. https://app.opensauced.pizza/u/JoshuaKGoldberg/card

CleanShot 2024-08-18 at 13 34 15

Note that the OG image is fine, as we don't add the OSCR score beside the username.

### PR DESCRIPTION
**Pull request by Mandeep56Singh**
## Description

This PR addresses the bug where a long username in a developer card causes the OSCR score to overflow and pop out of the card's boundary while also address related to it.

- resolve Hydaration error in Devcard component by using useEffect and useState for props.isLoading.
- fix styling in front view of card by limiting its width,text-center and wrapping
- fix styling in back view of card by truncate tailwind utility
- increase font size of oscr  limit  from 0.5 rem to 0.7rem to make it visible

closes #3981 

### Front view  at max username length
Screenshot (438)

### Back view at lengthy username
Screenshot (435)

### username in next line 
Screenshot (437)

## Tier (staff will fill in)

- [ ] Tier 1
- [ ] Tier 2
- [ ] Tier 3
- [ ] Tier 4

### CODE REVIEW

**Comment by nickytonline** [comment]
I was going to suggest not showing the OSCR on the back of the card since it's already on the front and leave as much real estate for the username. Thoughts @brandonroberts @zeucapua @isabensusan?

**Comment by Mandeep56Singh** [comment]
@brandonroberts  actually I had hydration error while solving this issue, To my understanding It was due to props.isLoading. 
However I have fixed it. I would require your assistance on this.

**Comment by brandonroberts** [comment]
> I was going to suggest not showing the OSCR on the back of the card since it's already on the front and leave as much real estate for the username. Thoughts @brandonroberts @zeucapua @isabensusan?

I agree

**Comment by BekahHW** [comment]
@nickytonline we have it on the back of the card like that to provide more context - it's out of 300. I think if we remove it here we need to have it referenced somewhere else on the card.

**Comment by nickytonline** [comment]
> @nickytonline we have it on the back of the card like that to provide more context - it's out of 300. I think if we remove it here we need to have it referenced somewhere else on the card.

Maybe we move it somewhere else on the card on the back. It looks nice to see a full username on the back instead of it being cut off.

**Comment by Mandeep56Singh** [comment]
@BekahHW we can it in front, we just need to add a line displaying 
there are various ways for it 
 
 ### we can display the out of message under score, but when we have long username ( 2 lines ) then it will look ugly
Screenshot (442)

### displaying it with OSCR score
Screenshot (440)

### I found this good, since we don't really need "score" in text as OSCR itself is rating 
Screenshot (443)

**Comment by Mandeep56Singh** [comment]
@nickytonline  full username on the back can take a lot of space as longest username can be of 39 characters, it would require two lines. I think  adding  more statistics on the back side, so user can know more about the person. We can remove the username, profile pic and oscr score from back. 

we have different option for it.
###  added new statistics and remove header of back view
Screenshot (444)

### this looks good
Screenshot (445)

what are your thoughts ?

**Comment by BekahHW** [comment]
I would prefer the original fix here. It solves the initial problem without a redesign. Maybe @isabensusan has thoughts.

**Comment by isabensusan** [comment]
@Mandeep56Singh I also prefer the original original fix! Let's truncate longer user names

**Comment by Mandeep56Singh** [comment]
@isabensusan  ok, the front view also have overflowing problem, the longest github username can be 39 characters long, so it will be two lines, I thought of centering the username. 
what you are your thoughts ?
thanks.

Screenshot (437)
Screenshot (438)

**Comment by isabensusan** [comment]
hey @Mandeep56Singh ! Yes, centering looks good  :)

**Comment by Mandeep56Singh** [comment]
> hey @Mandeep56Singh ! Yes, centering looks good :)

Thanks a lot, @isabensusan! I really appreciate the feedback. 
Looking forward to making more contributions to OpenSauced!"

**Review by isabensusan — APPROVED** [review]
Design wise, looks good to me!

File: components/molecules/DevCard/dev-card.tsx [hunk 1]
```diff
@@ -38,6 +39,13 @@ export default function DevCard(props: DevCardProps) {
     }
   }, [props.isInteractive, props.isFlipped]);
 
+  // used to solve hydartion error
+  useEffect(() => {
```
Comment (brandonroberts) [inline components/molecules/DevCard/dev-card.tsx:43]: We have a hook for this already named `useHasMounted` that can be used instead.

```ts
import { useHasMounted } from "lib/hooks/useHasMounted";

const hasMounted = useHasMounted();
```

**Review by brandonroberts — DISMISSED** [review]

File: components/molecules/DevCard/dev-card.tsx [hunk 2]
```diff
@@ -186,10 +190,10 @@ export default function DevCard(props: DevCardProps) {
                 height={50}
                 className="absolute rounded-full border-[1.5px] border-gray-300 border-opacity-40 -left-1"
               />
-              <p className="pl-10">{props.devstats?.login}</p>
+              <p className="pl-10 truncate items-center">{props.devstats?.login}</p>
               <p className="-pl-4 justfy-self-end text-sm flex gap-0.5">
```
Comment (nickytonline) [inline components/molecules/DevCard/dev-card.tsx:194]: Center the text. Also it looked like it was never self justifying to the end.

```suggestion
              <p className="-pl-4 justify-self-end items-center text-sm flex gap-0.5">
```

**Review by nickytonline — CHANGES_REQUESTED** [review]
Works great @Mandeep56Singh! One small change request and this is good to go.

CleanShot 2024-09-18 at 08 41 44

**Review by nickytonline — APPROVED** [review]
Thanks for making the changes @Mandeep56Singh! 🚢

**Review by brandonroberts — APPROVED** [review]

**Comment by nickytonline** [comment]
When you have a chance @zeucapua, we need your approval. Thanks!

**Pull request merged**